src/main/java/com/google/starfish/servlets/UserRegistrationServlet.java

```diff
@@ -0,0 +1,170 @@
+package com.google.starfish.servlets;  
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import java.io.IOException;  
+import java.sql.Connection;  
+import java.sql.SQLException;  
+import java.sql.PreparedStatement;
+import java.sql.Types;  
+import java.sql.ResultSet;
+import java.sql.Date;
+import javax.servlet.ServletException;  
+import java.security.GeneralSecurityException;
+import javax.servlet.annotation.WebServlet;  
+import javax.servlet.http.HttpServlet;  
+import javax.servlet.http.HttpSession;  
+import javax.servlet.http.Cookie;  
+import javax.servlet.http.HttpServletRequest;  
+import javax.servlet.http.HttpServletResponse;  
+import javax.sql.DataSource;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/** Servlet that handles user registration and authentication. */  
+@WebServlet("/user-registration")  
+public class UserRegistrationServlet extends HttpServlet {
+
+  private static final Logger LOGGER = Logger.getLogger(UserRegistrationServlet.class.getName());
+  private static final String CLIENT_ID = System.getenv("CLIENT_ID");
+  private final String COOKIE_NAME = "SFCookie";
+
+  @Override 
+  public void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    Cookie[] cookies = req.getCookies();
+    String sessionId = null;
+    for (Cookie cookie : cookies) {
```

Comment:
Similar to above suggestion, I recommend finding the cookie, caching the value in a local variable, and then using it.

Reply:
Sounds good.